src/auto/dbus_connection.rs

```diff
@@ -0,0 +1,1264 @@
+        reply_type: Option<&glib::VariantTy>,
+        flags: DBusCallFlags,
+        timeout_msec: i32,
+        fd_list: Option<&P>,
+        cancellable: Option<&Q>,
+    ) -> Result<(glib::Variant, UnixFDList), glib::Error> {
+        unsafe {
+            let mut out_fd_list = ptr::null_mut();
+            let mut error = ptr::null_mut();
+            let ret = gio_sys::g_dbus_connection_call_with_unix_fd_list_sync(
+                self.to_glib_none().0,
+                bus_name.to_glib_none().0,
+                object_path.to_glib_none().0,
+                interface_name.to_glib_none().0,
+                method_name.to_glib_none().0,
+                parameters.to_glib_none().0,
+                reply_type.to_glib_none().0,
+                flags.to_glib(),
+                timeout_msec,
+                fd_list.map(|p| p.as_ref()).to_glib_none().0,
+                &mut out_fd_list,
+                cancellable.map(|p| p.as_ref()).to_glib_none().0,
+                &mut error,
+            );
+            if error.is_null() {
+                Ok((from_glib_full(ret), from_glib_full(out_fd_list)))
+            } else {
+                Err(from_glib_full(error))
+            }
+        }
+    }
+
+    pub fn close<P: IsA<Cancellable>, Q: FnOnce(Result<(), glib::Error>) + Send + 'static>(
+        &self,
+        cancellable: Option<&P>,
+        callback: Q,
+    ) {
+        let user_data: Box_<Q> = Box_::new(callback);
+        unsafe extern "C" fn close_trampoline<
+            Q: FnOnce(Result<(), glib::Error>) + Send + 'static,
+        >(
+            _source_object: *mut gobject_sys::GObject,
+            res: *mut gio_sys::GAsyncResult,
+            user_data: glib_sys::gpointer,
+        ) {
+            let mut error = ptr::null_mut();
+            let _ =
+                gio_sys::g_dbus_connection_close_finish(_source_object as *mut _, res, &mut error);
+            let result = if error.is_null() {
+                Ok(())
+            } else {
+                Err(from_glib_full(error))
+            };
+            let callback: Box_<Q> = Box_::from_raw(user_data as *mut _);
+            callback(result);
+        }
+        let callback = close_trampoline::<Q>;
+        unsafe {
+            gio_sys::g_dbus_connection_close(
+                self.to_glib_none().0,
+                cancellable.map(|p| p.as_ref()).to_glib_none().0,
+                Some(callback),
+                Box_::into_raw(user_data) as *mut _,
+            );
+        }
+    }
+
+    pub fn close_future(
+        &self,
+    ) -> Pin<Box_<dyn std::future::Future<Output = Result<(), glib::Error>> + 'static>> {
+        Box_::pin(crate::GioFuture::new(self, move |obj, send| {
+            let cancellable = Cancellable::new();
+            obj.close(Some(&cancellable), move |res| {
+                send.resolve(res);
+            });
+
+            cancellable
+        }))
+    }
+
+    pub fn close_sync<P: IsA<Cancellable>>(
```

Comment:
I'm curious about this one: how is it different from the `close` method?

Reply:
The non-sync ones are async functions.